### ISSUE TITLE
erf - visual - allwords

### ISSUE THREAD

**Issue by schoffelen** [schoffelen/MOUS#21]
I suggest to work towards consistently running and documenting each type of analysis.
This will be kept track of in an issue, and attached to the milestone 'spring evaluation'

**Comment by annhul** [comment]
@schoffelen you made a "bugfix" in the mous_erf_pipline stating that the triggers were swapped between rc and mix, but I had another look at the presentation code and triggers 1-4 are RC and 5-8 are mix (granted the comments for the seq are misleading, but the code is correct). Logic of the code and stimuli goes as follows
RC sentences are stimuli # 1-204 they have trigger 1,2
MIX sentences are stimuli # 205-704 have triggers 5,6

RC sequences are stimuli 501 - 704 have triggers 3,4
MIX sequences are stimuli 706-908 have triggers 7,8

What made you think otherwise? Have I missed a mistake somewhere else?

**Comment by schoffelen** [comment]
Discusse (16 Feb): it seems to be a non-issue. The code seems correct

**Comment by annhul** [comment]
@schoffelen I get the following error when trying to rerun the rc/mix allwords with mne do_main:

Too many outputs requested.  Most likely cause is missing [] around left hand side
that has a comma separated list expansion.

Error in mous_mne_pipeline (line 171)
      lf   = sourcemodel.leadfield{indx};

I get what the error is (the indx is zero at those locations that we are not using) but I don't know how to fix it so that it wont cause more problems further down the code.

**Comment by schoffelen** [comment]
This is due to a recent change in FT, see http://mailman.science.ru.nl/pipermail/fieldtrip/2015-February/008903.html
I have updated the code (but did not test it).
Please git pull and try again (and keep your eyes open for other occurrences of this issue).

**Comment by annhul** [comment]
Did a git pull and got a new error: 
Error using  + 
Not enough input arguments.

Error in mous_mne_pipeline (line 224)
    mom = (source.avg.mom{source.inside(k)} +
    source_seq.avg.mom{source.inside(k)})./2;

**Comment by schoffelen** [comment]
Same cause.
Fixed it.
Please pull and be inspired by the way it was fixed (so that you can do it yourself if you get stuck).
Going offline now.

**Comment by annhul** [comment]
I get a error when trying to run the cluster correction for mcp (using the fdr correction works fine). The code is as follows: 
cfg          = [];
cfg.correctm = 'cluster';
cfg.numrandomization = 100;
stat_corr = mous_mne_groupanalysis(subj, suffix{1},'','parameter','avg.dspm','cfg',cfg,'sent',sent,'seq',seq);

%% And the error as such: 
Error using spm_bwlabel
spm_bwlabel: CONN must be 6, 18 or 26

Error in clusterstat (line 184)
      [posclusobs, posnum] = spm_bwlabel(tmp, 2*numdims);

Error in ft_statistics_montecarlo (line 356)
  [stat, cfg] = clusterstat(cfg, statrand, statobs);

Error in ft_sourcestatistics (line 202)
  [stat, cfg] = statmethod(cfg, dat, design);

Error in mous_mne_groupanalysis (line 171)
stat = ft_sourcestatistics(cfg,sent{:},seq{:});

My debugging got me as far as thinking that there might be something missing in the definition of which vertices are neighbours, and therefore the dimensions determining the connectivity fails...if this is correct I nevertheless do not know how to fix it. Thank you for your help!

**Comment by annhul** [comment]
I debugged a little further.. in clusterstat.m the variable numdims is optionally either 2 or 3 depending on if it is 2D or 3D data. The numdims variable is then multiplied by 2 and sent to  spm_bwlabel. However, if the data is 2D (as our data is),then 2*2 is 4 which is not 6, 18 or 26 as required by spm_bwlabel...so it seems to be a ft bug (?). Please advise @schoffelen as to how to proceed :)

**Comment by schoffelen** [comment]
It looks as if the code takes a wrong turn, it should not end up in that part to begin with.
Before I dig into this: your git-repository and fieldtrip are up to date?

**Comment by annhul** [comment]
Yes my git is up to date and ft is added by addpath /home/common/matlab/fieldtrip whereafter 
ft_defaults is run (as far as I know this runs the most recent version of ft...)

**Comment by schoffelen** [comment]
what script are you running. I don't know what suffix{1} is, and also not what you use as sent and seq variables.

**Comment by annhul** [comment]
/project/3011020.09/annhul/analyses/analysis20151228_1.m (the number of randomizations is low on purpose for debugging/test-running the code).

**Comment by schoffelen** [comment]
I don't know whether this used to work, but indeed the current version of the FT-code takes a wrong turn somewhere, not using the cfg.tri for the creation of the spatial neighborhood-matrix. I updated the fieldtrip code (in ft_statistics_montecarlo): it should work now.

**Comment by annhul** [comment]
Thank you, yes now it works! I am not sure it worked before either, which is may explain why I ended up using the fdr correction instead.

**Comment by jingjiangpsy** [comment]
Hi @schoffelen,  I used  2 versions of fieldtrip (fieldtrip-20160320 and fieldtrip-20160126) for source statistical analysis, but it still reports the same error as below after computing statistic 1000 from 1000.  The data are based on stc files created by mne_make_movie. 

I am wondering in these 2 versions, whether the ft_statistics_montecarlo you mentioned is updated or not. If updated, do you know other possible reasons resulting to this error?

-------------------------------------%% run statistics over subjects %
cfg=[];
cfg.method      = 'montecarlo';
cfg.statistic   = 'ft_statfun_depsamplesT';
cfg.parameter   = 'nai'; % 2 D data [N_sources x  timepoints]
cfg.correctm    = 'cluster';
cfg.numrandomization = 1000;
cfg.alpha       = 0.05; % note that this only implies single-sided testing
cfg.tail        = 0;

nsubj=length(sublist);
cfg.design(1,:) = [1:nsubj 1:nsubj];
cfg.design(2,:) = [ones(1,nsubj) ones(1,nsubj)*2];
cfg.uvar        = 1; % row of design matrix that contains unit variable (in this case: subjects)
cfg.ivar        = 2; % row of design matrix that contains independent variable (the conditions)

stat = ft_sourcestatistics(cfg, allsubjA{:}, allsubjB{:});

--------------------------------%% Error as such
using "ft_statfun_depsamplesT" for the single-sample statistics
constructing randomized design
total number of measurements     = 48
total number of variables        = 2
number of independent variables  = 1
number of unit variables         = 1
number of within-cell variables  = 0
number of control variables      = 0
using a permutation resampling approach
repeated measurement in variable 1 over 24 levels
number of repeated measurements in each level is 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 
computing a parametric threshold for clustering
computing statistic
estimated time per randomization is 2.86 seconds
computing statistic 1000 from 1000

Error using spm_bwlabel
spm_bwlabel: CONN must be 6, 18 or 26

Error in clusterstat (line 184)
      [posclusobs, posnum] = spm_bwlabel(tmp, 2*numdims);

Error in ft_statistics_montecarlo (line 347)
  [stat, cfg] = clusterstat(cfg, statrand, statobs);

Error in ft_sourcestatistics (line 205)
  [stat, cfg] = statmethod(cfg, dat, design);

Error in source_fieldtrip_statistic_test (line 86)
stat = ft_sourcestatistics(cfg, allsubjA{:},
allsubjB{:});
------------------------------

**Comment by schoffelen** [comment]
I just removed a comment of a person unknown to me, and this comment was not relevant for this thread. 
I am surprised actually that the comment ended up here to begin with.
